src/machine-learning/machine-learning-api-client.ts

```diff
@@ -140,6 +152,31 @@ export class MachineLearningApiClient {
       });
   }
 
+  public listModels(options: ListModelsOptions = {}): Promise<ListModelsResponse> {
+    if (typeof options.filter !== 'undefined' && !validator.isNonEmptyString(options.filter)) {
```

Comment:
Best to not rely on default value in the method signature. Otherwise an explicit `null` value may behave unexpectedly.

```
if (tyepeof options !== 'undefined') {
  if (!options.isNonNullObject(options)) {
    // reject
  }
}
```

Reply:
done (minus the 'undefined' that we discussed)